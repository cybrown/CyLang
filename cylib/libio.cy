void extern putchar(int c);
void extern putbyte(byte c);
void extern putshort(short c);
void extern putint(int c);
void extern putlong(long c);
void extern putfloat(float f);
void extern putdouble(double d);
void extern puts(byte[] str);
void extern printf(byte[] str);

void print(byte c) {
	putbyte(c);
}

void print(short c) {
	putshort(c);
}

void print(int c) {
	putint(c);
}

void print(long c) {
	putlong(c);
}

void print(float f) {
	putfloat(f);
}

void print(double d) {
	putdouble(d);
}

void print(byte[] str) {
	printf(str);
}

void print(bool c) {
	print(c ? "true" : "false");
}
