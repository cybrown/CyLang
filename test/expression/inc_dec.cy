import libio;

int a = 0;
putint(a++); putchar(10);
putint(a--); putchar(10);
putint(a--); putchar(10);

putint(++a); putchar(10);
putint(--a); putchar(10);
putint(++a); putchar(10);

float f = 5.0f;
double d = 5.0d;
putfloat(f++); putchar(10);
putfloat(f--); putchar(10);
putfloat(f++); putchar(10);

putdouble(--d); putchar(10);
putdouble(++d); putchar(10);
putdouble(--d); putchar(10);