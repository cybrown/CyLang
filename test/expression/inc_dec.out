0
1
0
0
-1
0
5.000000
6.000000
5.000000
4.000000
5.000000
4.000000
